### PR TITLE
Add consts.rs, download_util.rs. Move some code

### PR DESCRIPTION
**Pull request by suprohub**

### CODE REVIEW

File: crates/nomi-core/src/download/assets.rs [hunk 1]
```diff
@@ -51,16 +51,6 @@ impl AssetsDownload {
         Ok(data)
     }
 
-    async fn create_dir(&self, main_dir: &Path, asset_dir_name: &str) -> Result<PathBuf> {
```
Comment (Umatriz) [inline crates/nomi-core/src/download/assets.rs:54]: You dont need to move this function away. Leave it inside `Assets` implementation block

File: crates/nomi-core/src/lib.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 pub mod configs;
-pub mod downloads;
+pub mod download;
```
Comment (Umatriz) [inline crates/nomi-core/src/lib.rs:2]: Whats wrong with `downloads`?

File: .vscode/settings.json
Comment (Umatriz) [inline .vscode/settings.json:1]: Idk what `rust-analyzer.linkedProjects` does. If its something useful pls share the docs link

File: crates/nomi-core/src/configs/consts.rs
Comment (Umatriz) [inline crates/nomi-core/src/configs/consts.rs:1]: move this into `ustils::consts`

File: crates/nomi-core/src/utils/download_util.rs
Comment (Umatriz) [inline crates/nomi-core/src/utils/download_util.rs:1]: You dont need to move `download_file` into this file. Leave it in the `downloads::mod.rs`

File: crates/nomi-core/src/utils/download_util.rs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+use tokio::io::AsyncWriteExt;
+use tracing::{debug, error, trace};
+
+#[derive(Error, Debug)]
+enum DownloadManagerError {
+    #[error("Hash does not match")]
+    HashDoesNotMatch,
+}
+
+pub(crate) async fn check_sha256(path: PathBuf, hash: &str) -> anyhow::Result<()> {
```
Comment (Umatriz) [inline crates/nomi-core/src/utils/download_util.rs:14]: This `check_sha256` used only in `JavaInstalled`. Put it back into `JavaInstalled` impl block and make in privat

File: crates/nomi-core/src/configs/consts.rs
Comment (suprohub) [reply crates/nomi-core/src/configs/consts.rs:1]: ok

File: .vscode/settings.json
Comment (suprohub) [reply .vscode/settings.json:1]: i forgot add gitignore

File: crates/nomi-core/src/download/assets.rs (hunk 1 above)
Comment (suprohub) [reply crates/nomi-core/src/download/assets.rs:54]: ok

File: crates/nomi-core/src/utils/download_util.rs (hunk 3 above)
Comment (suprohub) [reply crates/nomi-core/src/utils/download_util.rs:14]: ok

**Pull request closed without merging**